### DISCUSSION TITLE
Is there a get<std::string>() method for lists?

### DISCUSSION THREAD

**Discussion by Skuttle-Studios** [nlohmann/json#3877] [Q&A] [answered]
I'm trying to get a list of `std::string`s from an list of JSON strings so that I can loop through them. Is there a version of the get method that does that? I used `stringlist.get<std::string>()` but it said that the type isn't a string(actually it tried to but then threw an error saying the exception didn't work either). This led me to the assumption that it doesn't accept lists. So, once again, is there a version of `get<std::string>()` for lists?

Sorry for the confusing question, and I'm using Visual Studio

EDIT: an example of a "std::string list" is something like ["This", "is", "a", "list", "of", "strings"]. Now if I wanted to edit that but still be able to loop through and print out(using cout <<) each item as an std::string

**Comment by nlohmann** [comment, marked as answer]
Can you provide more details like a running example or the code that is failing?

**Reply by Skuttle-Studios** [reply]
> the code that is failing

The internal JSON code that's producing an error or my projects code?

**Reply by nlohmann** [reply]
I don't know what "a list of std::string" is exactly or what "a list of JSON strings" is. Please provide a minimal example.

**Reply by Skuttle-Studios** [reply]
A list like ["hello", "apple"]

**Reply by Skuttle-Studios** [reply]
sorry for the confusion

**Reply by nlohmann** [reply]
Assuming you have a JSON array of strings:

```cpp
json j = {"foo", "bar", "baz"};
```

then you can do the following:

```cpp
auto l = j.get<std::list<std::string>>();
```

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Comment by Skuttle-Studios** [comment]
I do not believe so. Thank you so much!
